source/Line.hpp

```diff
@@ -95,12 +97,39 @@ class Line : public io::IO
 	 * @param curv The curvature
 	 */
 	real getNonlinearEI(real curv);
+	
+	/** @brief Finds the depth of the water at some (x, y) point. Either using env->WtrDpth or the 3D seafloor if available
+	 * @param x x coordinate
+	 * @param y y coordinate
+	 * @return A negative number representing the sea floor depth at the given location
+	*/
+	real getWaterDepth(real x, real y) {
+		if (seafloor) {
+			return seafloor->getDepthAt(x, y);
+		}
+		else {
+			return -env->WtrDpth;
+		}
+	}
 
+	/** @brief A single value representing the average water depth
+	 * 
+	*/
+	real avgWaterDepth() {
+		if (seafloor) {
```

Comment:
You can use an inline if:

`return seafloor ? seafloor->getAverageDepth() : -env->WtrDpth;`

Up to you!

Reply:
I guess this gets to the great ternary vs. classic conditional readability debate. I also prefer the oneliner for things like this so I'll make that update.